packages/commons/src/rate-limiter/redisRateLimiter.ts

```diff
@@ -0,0 +1,130 @@
+  maxRequests: number;
+  rateInterval: number;
+  burstPercentage: number;
+  redisHost: string;
+  redisPort: number;
+  timeout: number;
+}): Promise<RateLimiter> {
+  const redisClient = await createRedisClient({
+    socket: {
+      host: config.redisHost,
+      port: config.redisPort,
+      connectTimeout: config.timeout,
+    },
+  })
+    .on("error", (err) => genericLogger.warn(`Redis Client Error: ${err}`))
+    .connect();
+
+  const options: IRateLimiterRedisOptions = {
+    storeClient: redisClient,
+    keyPrefix: config.limiterGroup,
+    points: config.maxRequests,
+    duration: config.rateInterval / 1000, // seconds
+  };
+
+  const burstOptions: IRateLimiterRedisOptions = {
+    ...options,
+    keyPrefix: `${burstKeyPrefix}${config.limiterGroup}`,
+    points: config.maxRequests * config.burstPercentage,
+    duration: (config.rateInterval / 1000) * config.burstPercentage,
+  };
+
+  const rateLimiter = new BurstyRateLimiter(
+    new RateLimiterRedis(options),
+    new RateLimiterRedis(burstOptions)
+  );
+  // ^ The BurstyRateLimiter is a RateLimiter that allows traffic bursts that exceed the rate limit.
+  // See: https://github.com/animir/node-rate-limiter-flexible/wiki/BurstyRateLimiter
+
+  async function rateLimitByOrganization(
+    organizationId: TenantId,
+    logger: Logger
+  ): Promise<RateLimiterStatus> {
+    try {
+      const rateLimiterRes = await rateLimiter.consume(organizationId);
+      return {
+        limitReached: false,
+        maxRequests: config.maxRequests,
+        remainingRequests: rateLimiterRes.remainingPoints,
+        rateInterval: config.rateInterval,
+      };
+    } catch (error) {
+      return match(error)
+        .with(P.instanceOf(RateLimiterRes), (rejRes) => {
+          logger.warn(
+            `Rate Limit triggered for organization ${organizationId}`
+          );
+          return {
+            limitReached: true,
+            maxRequests: config.maxRequests,
+            remainingRequests: rejRes.remainingPoints,
+            rateInterval: config.rateInterval,
+          };
+        })
+        .with(P.intersection(P.instanceOf(ConnectionTimeoutError)), () => {
+          logger.warn(
+            `Redis command timed out, making request pass for organization ${organizationId}`
+          );
+          return {
+            limitReached: false,
+            maxRequests: config.maxRequests,
+            remainingRequests: config.maxRequests,
+            rateInterval: config.rateInterval,
+          };
+        })
+        .otherwise((error) => {
+          logger.warn(
+            `Unexpected error during rate limiting for organization ${organizationId} - ${error}`
+          );
+          return {
+            limitReached: false,
+            maxRequests: config.maxRequests,
+            remainingRequests: config.maxRequests,
+            rateInterval: config.rateInterval,
+          };
+        });
+    }
+  }
+
+  async function getCountByOrganization(
```

Comment:
I'm probably missing it, but is this function used?

Reply:
It's used only in the tests, but since it's just a read I thought it was ok to implement it even though only for testing purposes